pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -176,6 +176,15 @@ func (c *Cluster) Start(clusterInfo *cephconfig.ClusterInfo, rookVersion string,
 		return nil, fmt.Errorf("failed to initialize ceph cluster info. %+v", err)
 	}
 
+	availableNodes, _, err := c.getAvailableMonNodes()
```

Comment:
`getAvailableMonNodes()` will return a different set of available nodes if there are already mons running. For example, if you have 1 mon running, then change the mon count to 3, it will look like you have one node fewer available where the mon is running. I think what we want is a new helper method to be implemented based on [these lines](https://github.com/rook/rook/blob/233c3211034fe52355ae3bd34deb16dd7c6e8e29/pkg/operator/ceph/cluster/mon/node.go#L118-L126) that would return the total set of available nodes.  Then `getTargetMonCount()` could also call this helper.

Reply:
Thanks for your review, i'll address it.